src/Symfony/Component/PropertyInfo/Type.php

```diff
@@ -11,27 +11,38 @@
 
 namespace Symfony\Component\PropertyInfo;
 
+use Symfony\Component\TypeInfo\BackwardCompatibilityHelper;
+use Symfony\Component\TypeInfo\Type as TypeInfoType;
+use Symfony\Component\TypeInfo\Type\CollectionType;
+use Symfony\Component\TypeInfo\Type\GenericType;
+use Symfony\Component\TypeInfo\Type\ObjectType;
+use Symfony\Component\TypeInfo\TypeIdentifier;
+
+trigger_deprecation('symfony/property-info', '7.1', 'The "%s" class is deprecated. Use "%s" of "symfony/type-info" component instead.', Type::class, TypeInfoType::class);
+
 /**
  * Type value object (immutable).
  *
  * @author Kévin Dunglas <dunglas@gmail.com>
  *
+ * @deprecated since Symfony 7.1, use "Symfony\Component\TypeInfo\Type" of "symfony/type-info" component instead
+ *
  * @final
  */
 class Type
 {
-    public const BUILTIN_TYPE_INT = 'int';
-    public const BUILTIN_TYPE_FLOAT = 'float';
-    public const BUILTIN_TYPE_STRING = 'string';
-    public const BUILTIN_TYPE_BOOL = 'bool';
-    public const BUILTIN_TYPE_RESOURCE = 'resource';
-    public const BUILTIN_TYPE_OBJECT = 'object';
-    public const BUILTIN_TYPE_ARRAY = 'array';
-    public const BUILTIN_TYPE_NULL = 'null';
-    public const BUILTIN_TYPE_FALSE = 'false';
-    public const BUILTIN_TYPE_TRUE = 'true';
-    public const BUILTIN_TYPE_CALLABLE = 'callable';
-    public const BUILTIN_TYPE_ITERABLE = 'iterable';
+    public const BUILTIN_TYPE_INT = TypeIdentifier::INT->value;
+    public const BUILTIN_TYPE_FLOAT = TypeIdentifier::FLOAT->value;
+    public const BUILTIN_TYPE_STRING = TypeIdentifier::STRING->value;
+    public const BUILTIN_TYPE_BOOL = TypeIdentifier::BOOL->value;
+    public const BUILTIN_TYPE_RESOURCE = TypeIdentifier::RESOURCE->value;
+    public const BUILTIN_TYPE_OBJECT = TypeIdentifier::OBJECT->value;
+    public const BUILTIN_TYPE_ARRAY = TypeIdentifier::ARRAY->value;
+    public const BUILTIN_TYPE_NULL = TypeIdentifier::NULL->value;
+    public const BUILTIN_TYPE_FALSE = TypeIdentifier::FALSE->value;
+    public const BUILTIN_TYPE_TRUE = TypeIdentifier::TRUE->value;
+    public const BUILTIN_TYPE_CALLABLE = TypeIdentifier::CALLABLE->value;
+    public const BUILTIN_TYPE_ITERABLE = TypeIdentifier::ITERABLE->value;
```

Comment:
not sure this change is needed, I'd prefer reverting it so make it clear that we're keeping the same values (do we?)